ouroboros-consensus/src/Ouroboros/Consensus/Storage/ImmutableDB/Impl/Validation.hs

```diff
@@ -501,3 +508,71 @@ reconstructPrimaryIndex chunkInfo HashInfo { hashSize } shouldBeFinalised
                                   + Secondary.entrySize hashSize
               in backfilled ++ secondaryOffset
                : go (nextRelativeSlot relSlot) secondaryOffset relSlots'
+
+
+{------------------------------------------------------------------------------
+  Migration
+------------------------------------------------------------------------------}
+
+-- | Migrate the files in the database to the latest version.
+--
+-- We always migrate the database to the latest version before opening it. If
+-- a migration was unsuccessful, an error is thrown and the database is not
+-- opened. User intervention will be needed before the database can be
+-- reopened, as without it, the same error will be thrown when reopening the
+-- database the next time.
+--
+-- For example, when during a migration we have to rename a file A to B, but
+-- we don't have permissions to do so, we require user intervention.
+--
+-- We have the following versions, from current to oldest:
+--
+-- * Current version:
+--
+--   - Chunk files are named "XXXXX.chunk" where "XXXXX" is the chunk/epoch
+--     number padded with zeroes to five decimals. A chunk file stores the
+--     blocks in that chunk (which coincides with epochs for Byron)
```

Comment:
If we choose chunk size to be equal to the Byron epoch size and allow for EBBs, the contents of these files doesn't have to change.